tests/qos/test_buffer.py

```diff
@@ -1888,3 +1959,189 @@ def test_buffer_model_test(duthosts, rand_one_dut_hostname, conn_graph_facts):
+        return False, None
+
+    # Get pause_quanta with port speed from pause_quanta_per_speed_dict
+    if port_speed in pause_quanta_per_speed_dict.keys():
+        pause_quanta = pause_quanta_per_speed_dict[port_speed]
+    else:
+        pause_quanta = 60
+
+    # Get port mtu from config DB
+    # Command: redis-cli -n 4 hget "PORT|Ethernet0" 'mtu'
+    port_mtu_raw = duthost.shell('redis-cli -n 4 hget "PORT|{}" "mtu"'.format(port_to_test))['stdout']
+    if port_mtu_raw:
+        port_mtu = int(port_mtu_raw)
+    else:
+        logging.error("failed to get MTU from config db for port {}".format(port_to_test))
+        return False, None
+
+    # Determine gearbox_delay with platform name, so far only MSN3800 has gear_box installed
+    if duthost.facts["platform"] not in ["x86_64-mlnx_msn3800-r0"]:
+        gearbox_delay = 0
+    else:
+        gearbox_delay_keys = duthost.shell('redis-cli -n 6 keys PERIPHERAL_TABLE*')['stdout']
+        gearbox_delay = float(duthost.shell('redis-cli -n 6 hget "{}" "gearbox_delay"'.format(gearbox_delay_keys))['stdout'])
+
+    # Get cable length from config DB
+    # Command: redis-cli -n 4 hget "CABLE_LENGTH|AZURE"  'Ethernet0'
+    cable_length_keys = duthost.shell('redis-cli -n 4 keys *CABLE_LENGTH*')['stdout']
+    cable_length_raw = duthost.shell('redis-cli -n 4 hget "{}" "{}"'.format(cable_length_keys, port_to_test))['stdout']
+    if cable_length_raw and cable_length_raw.endswith('m'):
+        cable_length = float(cable_length_raw[:-1])
+    else:
+        logging.error("failed to get a valid cable length from config db for port {}".format(port_to_test))
+        return False, None
+
+    logging.info('port_speed = {}, port_mtu = {}, cable_length = {}'.format(port_speed, port_mtu, cable_length))
+
+    # Get port lanes number from config DB
+    # Command: redis-cli -n 4 hget "PORT|Ethernet0" 'lanes'
+    port_lanes = duthost.shell('redis-cli -n 4 hget "PORT|{}" "lanes"'.format(port_to_test))['stdout']
+    is_8lane = port_lanes and len(port_lanes.split(',')) == 8
+
+    if not ASIC_TABLE_KEYS_LOADED:
+        # Get cell size from state DB
+        # Command: redis-cli -n 6 hget "ASIC_TABLE|MELLANOX-SPECTRUM-2" 'cell_size'
+        asic_keys = duthost.shell('redis-cli -n 6 keys *ASIC_TABLE*')['stdout']
+        CELL_SIZE = float(duthost.shell('redis-cli -n 6 hget "{}" "cell_size"'.format(asic_keys))['stdout'])
+
+        # Get PIPELINE_LATENCY from state DB
+        # Command: redis-cli -n 6 hget "ASIC_TABLE|MELLANOX-SPECTRUM-2" 'pipeline_latency'
+        PIPELINE_LATENCY = float(duthost.shell('redis-cli -n 6 hget "{}" "pipeline_latency"'.format(asic_keys))['stdout']) * 1024
+
+        # Get MAC_PHY_DELAY from state DB
+        # Command: redis-cli -n 6 hget "ASIC_TABLE|MELLANOX-SPECTRUM-2" 'mac_phy_delay'
+        MAC_PHY_DELAY = float(duthost.shell('redis-cli -n 6 hget "{}" "mac_phy_delay"'.format(asic_keys))['stdout']) * 1024
```

Comment:
can this be moved to a function? since the same set of parameters are retrieved in detect_asic_keys as well

Reply:
added a new function "get_asic_table_data_from_db"